packages/ai-jsx/src/react/serialize.tsx

```diff
@@ -0,0 +1,30 @@
+/** @jsx AI.createElement */
+import * as AI from './index.js';
+import React from 'react';
+
+function reactComponentName(component: React.JSXElementConstructor<any> | string) {
+  return typeof component === 'string' ? component : component.name;
+}
+
+/**
+ * Serializes React components to a textual representation.
+ */
+export function Serialize({ children }: { children: React.ReactNode }): AI.Renderable {
+  if (Array.isArray(children)) {
+    return children.map((child) => <Serialize>{child}</Serialize>);
+  }
+
+  const child = children;
+  if (React.isValidElement(child) && child.type !== AI.jsx) {
+    // Serialize the React element and any children.
+    const typeName = reactComponentName(child.type);
+    // TODO: support prop serialization
```

Comment:
Yeah, and we don't support prop application either right now, right? That's fine; we should just call it out as a known caveat.

Reply:
I didn't end up adding anything just because I think this is something we almost certainly want to fix Soon